content/en/docs/concepts/architecture/controller.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Controllers
+content_template: templates/concept
+weight: 30
+---
+
+{{% capture overview %}}
+
+In robotics and automation, a _control loop_ is
+a non-terminating loop that regulates the state of a system.
+
+Here is one example of a control loop: a thermostat in a room.
+
+When you set the temperature, that's telling the thermostat
+about your *desired state*. The actual room temperature is the
+*current state*. The thermostat acts to bring the current state
+closer to the desired state, by turning equipment on or off.
+
+{{< glossary_definition term_id="controller" length="short">}}
+
+{{% /capture %}}
+
+
+{{% capture body %}}
+
+## Controller pattern
+
+A controller tracks at least one Kubernetes resource type.
+These [objects](/docs/concepts/overview/working-with-objects/kubernetes-objects/)
+and their associated data are the desired state, so the
```

Comment:
their spec field is the desired state.

Reply:
@cheftako are you sure? Does every resource _have_ a `spec` field, that completely describes the desired state?

Reply:
Yes. From our documentation (https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/) 

"Every Kubernetes object includes two nested object fields that govern the object’s configuration: the object spec and the object status. The spec, which you must provide, describes your desired state for the object–the characteristics that you want the object to have. The status describes the actual state of the object..."

Reply:
Hmm, yes. I hadn't realized that was supposed to apply to _every_ object, though I'd seen it as commonly done.
Some API objects don't have a `spec`, eg: Secret, ConfigMap, Role, VolumeSnapshotClass, StorageClass, and (at a pinch) AdmissionReview.

For the purposes of explaining controllers that might be a detail worth glossing over. On the other hand, readers like me might wonder what's going on with those other objects, especially since several of the ones I've listed are relevant to built-in controllers.

@cheftako what are your thoughts?

Reply:
Hunh, that is a interesting point.

My biggest thought is that object without a spec and status is not build to be "controlled" in the way we are discussing.
If the object does not have spec and status to track desired vs real state then the objects cannot be used for declarative control loops.
These objects seem more ?informative?. We should probably have a better explanation for this under kubernetes-objects but for control loops I think it makes sense to just deal with status + spec objects as they seem the relevant objects for control loops.

Reply:
I'll work on a revision that covers this. Thanks for the feedback @cheftako